### ISSUE TITLE
incompatible_preprocess_asm_files

### ISSUE THREAD

**Issue by emusand** [bazelbuild/bazel#8055]
Bazel will start preprocess .asm files. Today Bazel assumes that ".s" and ".asm" files are generated by a C/C++ compiler and do not need to be parsed by a C preprocessor. However, there are companies with large codebases in C and assembly which use ".asm" as file extension to handwritten assembly files with preprocessor directives.

Note that --[no]incompatible_preprocess_asm_files is a start-up option, and not a build option.

See #4856

**Migration**
Make sure that your ".asm" files can be preprocessed. This basically means that you should build your code with the --incompatible_preprocess_asm_files start-up option, if you have ".asm" files in your codebase.

**Expected timeline**
Released with Bazel 0.26 (June 2019), off by default
Enabled by default with Bazel 0.29 (September 2019)

**Comment by brandjon** [comment]
This flag does not appear to be available in Bazel 0.25, but it has the `migration-0.25` label, causing it to [run and fail](https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/104#5b8edad8-db18-440d-a0cb-08614a27c374/1360) in the bazelisk pipeline.

**Comment by emusand** [comment]
Thanks for the information @brandjon!

This flag was unfortunately delayed. I changed the expected release in the description from 0.25 to 0.26. Can someone help me removin the migration-0.25 label?

**Comment by hlopko** [comment]
Hi @emusand, is this flag migration ready (meaning it can be announced and tested on the CI)?

**Comment by emusand** [comment]
Hi @hlopko,

Unfortunately this flag is not migration ready yet, as the [commit](https://github.com/bazelbuild/bazel/pull/6919) that introduces it is not ready to be released.

I am mainly waiting for comments on if the way I implemented the flag is OK. I had to introdude the flag in the client, instead of in the server, which might be controversial. But if you think that flag is OK, I can rebase the commit and fix the failures, and make sure that the commit soon is ready.
